llvm/lib/Transforms/IPO/GlobalOpt.cpp

```diff
@@ -1901,6 +1902,24 @@ static void makeAllConstantUsesInstructions(Constant *C) {
   }
 }
 
+/// Reset the global varible GV TicketNode metadata.
+static void calculateGlobalDigest(GlobalVariable *GV) {
+  const Module *const M = GV->getParent();
+  if (!M)
+    return;
+
+  const std::string TripleString = M->getTargetTriple();
```

Comment:
Should probably be a reference rather than a copy of the string. Elsewhere I've seen something along the lines of the incantation: `Triple(M‑>getTargetTriple()).isOSBinFormatRepo())`. Could this be used to simplify the code?

Reply:
> 
> Elsewhere I've seen something along the lines of the incantation: `Triple(M‑>getTargetTriple()).isOSBinFormatRepo())`. Could this be used to simplify the code?
>
Good point. I have fixed following your advice. Thanks.